reconcile/unleash_watcher.py

```diff
@@ -68,40 +69,16 @@ def calculate_diff(current_state, previous_state):
     return diffs
 
 
-def init_slack_map(unleash_instance):
+def init_slack_map(unleash_instance) -> Dict[str, SlackApi]:
     settings = queries.get_app_interface_settings()
     slack_notifications = unleash_instance['notifications']['slack']
     slack_map = {}
     for slack_info in slack_notifications:
-        workspace = slack_info['workspace']
-        workspace_name = workspace['name']
-        slack_integrations = workspace['integrations']
-        client_config = workspace.get('api_client')
-        slack_config = \
-            [i for i in slack_integrations
-             if i['name'] == QONTRACT_INTEGRATION]
-        [slack_config] = slack_config
-
-        token = slack_config['token']
         channel = slack_info['channel']
-        icon_emoji = slack_info['icon_emoji']
-        username = slack_info['username']
-
-        slack_api_kwargs = {
-            'settings': settings,
-            'init_usergroups': False,
-            'channel': channel,
-            'icon_emoji': icon_emoji,
-            'username': username
-        }
-
-        if client_config:
-            slack_api_kwargs['api_config'] = \
-                SlackApiConfig.from_dict(client_config)
-
-        slack = SlackApi(workspace_name, token, **slack_api_kwargs)
-
-        slack_map[channel] = slack
+        slack_map[channel] = \
+            slackapi_from_slack_workspace(slack_info, settings,
+                                          QONTRACT_INTEGRATION,
+                                          init_usergroups=False)
```

Comment:
nite: since you got rid of everything in this loop but the core, you could also use a dict comprehension to build the slack_map, but this is more about taste and completely up to you